packages/autocertifier-server/src/RestServer.ts

```diff
@@ -0,0 +1,230 @@
+
+const logger = new Logger(module)
+
+type ExpressType = ReturnType<typeof express>
+type ServerType = ReturnType<ExpressType['listen']>
+
+const sendError = (res: express.Response, err: any) => {
+    if (err instanceof Err) {
+        logger.error('Error ' + JSON.stringify(err))
+        res.status(err.httpStatus).send(err.toApiError())
+    } else {
+        logger.error('Unspecified error ' + JSON.stringify(err))
+        const unspecifiedError = new UnspecifiedError('Unspecified error')
+        res.status(unspecifiedError.httpStatus).send(unspecifiedError.toApiError())
+    }
+}
+
+const sendResponse = (res: express.Response, data?: object) => {
+    if (!data) {
+        res.json({})
+    } else {
+        res.json(data)
+    }
+}
+
+const extractIpAndPort = (req: express.Request): { ip: string, port: string } | undefined => {
+    // take x-forwarded for into account
+    const remoteIp = req.headers['x-forwarded-for'] || req.socket.remoteAddress
+    const remotePort = req.headers['x-forwarded-port'] || req.socket.remotePort
+    let ip = remoteIp
+    let port = remotePort
+    if (typeof remoteIp !== 'string' && typeof remoteIp !== 'number') {
+        if (Array.isArray(remoteIp) && remoteIp.length > 0) {
+            ip = remoteIp[0]
+        } else {
+            logger.error('invalid remote ip: ' + remoteIp)
+            return undefined
+        }
+    }
+    if (typeof remotePort !== 'string' && typeof remotePort !== 'number') {
+        if (Array.isArray(remotePort) && remotePort.length > 0) {
+            port = remotePort[0]
+        } else {
+            logger.error('invalid remote port: ' + remotePort)
+            return undefined
+        }
+    }
+    logger.info('extracted ip: ' + ip + ' port: ' + port + ' from request')
+    return { ip: '' + ip, port: '' + port }
+}
+
+export class RestServer {
+
+    private server?: ServerType
+    private engine: RestInterface
+
+    private ownIpAddress: string
+    private port: number
+    private certPath: string
+    private keyPath: string
+
+    constructor(ownIpAddress: string, port: number, certPath: string, keyPath: string, engine: RestInterface) {
+        this.ownIpAddress = ownIpAddress
+        this.port = port
+        this.certPath = filePathToNodeFormat(certPath)
+        this.keyPath = filePathToNodeFormat(keyPath)
+        this.engine = engine
+    }
+
+    public async start(): Promise<void> {
+        return new Promise<void>((resolve, _reject) => {
+
+            const app = express()
+            app.use(bodyParser.json())
+
+            app.get('/robots.txt', (_req, res) => {
+                res.type('text/plain')
+                res.send('User-agent: *\nDisallow: /')
+            })
+
+            // create new session
+            app.post('/sessions', this.createSession)
+
+            // create new subdomain and certificate
+            app.patch('/certified-subdomains', async (req, res) => {await this.createSubdomainAndCertificate(req, res)})
```

Comment:
```suggestion
            app.patch('/certificates', async (req, res) => {await this.createSubdomainAndCertificate(req, res)})
```

(or `/subdomains`, but I guess the `certified` prefix is not needed as it does distinguish from other types of subdomains)